[DROOLS-6584] Expose Drools metrics via Micrometer (#3860)
kiegroup/kie-docs: doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/performance-tuning-drl-ref.adoc

</dependency>
----

+

> If you wish `drools-metric` to enable trace logging, configure a logger for `org.drools.metric.util.MetricLogUtils` as shown in the following example:

Reviewer: ```suggestion
If you want to use `drools-metric` to enable trace logging, configure a logger for `org.drools.metric.util.MetricLogUtils` as shown in the following example:
```

----

+

> Alternatively, you may have `drools-metric` expose the data via https://asciidoctor.org[Micrometer]. 

Reviewer: ```suggestion
Alternatively, you can use `drools-metric` to expose the data using https://asciidoctor.org[Micrometer]. To expose the data, enable the Micrometer registry of your choice as shown in the following example:
```

> In order to do that, you need to enable the Micrometer registry of your choice as shown in the following example:

Reviewer: ```suggestion

```

+
.Example project dependency for Micrometer
[source,xml]
----
<dependency>
  <groupId>io.micrometer</groupId>
  <artifactId>micrometer-registry-jmx</artifactId> <!-- Discover more registries at micrometer.io. -->
</dependency>
----

+
.Example Java code for Micrometer
[source,xml]
----
  Metrics.addRegitry(new JmxMeterRegistry(s -> null, Clock.SYSTEM));
----

+

> Finally, regardless of whether you want to use logging or Micrometer, you need to enable `MetricLogUtils` by setting the system property `drools.metric.logger.enabled` to `true`. Optionally, you can change the microseconds threshold of metric reporting by setting the `drools.metric.logger.threshold` system property.

Reviewer: ```suggestion
Regardless of whether you want to use logging or Micrometer, you need to enable `MetricLogUtils` by setting the system property `drools.metric.logger.enabled` to `true`. Optionally, you can change the microseconds threshold of metric reporting by setting the `drools.metric.logger.threshold` system property.
```

+
NOTE: Only node executions exceeding the threshold are reported. The default value is `500`.

+

> Assuming you've configured `drools-metric` to use logging, rule execution produces logs as shown in the following example:

Reviewer: ```suggestion
After configuring the `drools-metric` to use logging, rule execution produces logs as shown in the following example:
```

--
This example includes the following key parameters:

> * `evalCount` is the number of constraint evaluations against inserted facts during the node execution. When used with Micrometer, a counter with this data is called `org.drools.metric.evaluation.count`.

Reviewer: ```suggestion
* `evalCount` is the number of constraint evaluations against inserted facts during the node execution. When `evalCount` is used with Micrometer, a counter with the data is called `org.drools.metric.evaluation.count`.
```

> * `elapsedMicro` is the elapsed time of the node execution in microseconds. When used with Micrometer, look for a timer called `org.drools.metric.elapsed.time`.

Reviewer: ```suggestion
* `elapsedMicro` is the elapsed time of the node execution in microseconds. When `elapsedMicro` is used with Micrometer, look for a timer called `org.drools.metric.elapsed.time`.
```
